src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -45,7 +45,7 @@ internal sealed class NamedPipeConnectionListener : IConnectionListener
         // The OS maintains a backlog of clients that are waiting to connect, so the app queue only stores a single connection.
         // We want to have a queue plus a background task that populates the queue, rather than creating NamedPipeServerStream
         // when AcceptAsync is called, so that the server is always the owner of the pipe name.
-        _acceptedQueue = Channel.CreateBounded<ConnectionContext>(new BoundedChannelOptions(capacity: 1) { SingleWriter = true });
+        _acceptedQueue = Channel.CreateBounded<ConnectionContext>(new BoundedChannelOptions(capacity: 1));
```

Comment:
You'll probably want to increase the capacity to the same size as the number of accept loops.

Reply:
I measured and it doesn't make a difference.

Reply:
I gotta admit, I'm still having a hard time with this being 1. 16 cores all polling the named pipe, then they race trying to write to the channel, but then they all bottle neck on it?

Reply:
https://github.com/dotnet/aspnetcore/pull/46472#issuecomment-1418553905